src/test/java/org/elasticsearch/action/admin/cluster/repositories/verify/VerifyRepositoryTests.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.cluster.repositories.verify;
+
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.common.settings.ImmutableSettings;
+import org.elasticsearch.test.ElasticsearchIntegrationTest;
+import org.elasticsearch.test.ElasticsearchIntegrationTest.ClusterScope;
+import org.junit.Test;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertBlocked;
+import static org.hamcrest.Matchers.equalTo;
+
+@ClusterScope(scope = ElasticsearchIntegrationTest.Scope.TEST)
+public class VerifyRepositoryTests extends ElasticsearchIntegrationTest {
+
+    @Test
+    public void testVerifyRepositoryWithBlocks() {
+        Client client = client();
+
+        assertAcked(client.admin().cluster().preparePutRepository("test-repo-blocks")
+                .setType("fs")
+                .setVerify(false)
+                .setSettings(ImmutableSettings.settingsBuilder().put("location", newTempDirPath())));
+
+        logger.info("-->  repository verification is blocked when the cluster is read only");
+        try {
+            setClusterReadOnly(true);
+            assertBlocked(client.admin().cluster().prepareVerifyRepository("test-repo-blocks"), MetaData.CLUSTER_READ_ONLY_BLOCK);
+        } finally {
+            setClusterReadOnly(false);
+        }
+
+        logger.info("--> repository verification is allowed when the cluster is not read only");
+        VerifyRepositoryResponse response = client.admin().cluster().prepareVerifyRepository("test-repo-blocks").execute().actionGet();
+        assertThat(response.getNodes().length, equalTo(cluster().numDataAndMasterNodes()));
+    }
+}
```

Comment:
I wonder if we should merge all the above repository tests in `RepositoryTests` or something along those lines